core/endorser/endorser.go

```diff
@@ -368,6 +379,12 @@ func (e *Endorser) ProcessProposalSuccessfullyOrError(up *UnpackedProposal) (*pb
 		txParams.HistoryQueryExecutor = hqe
 	}
 
+	// Throttle the concurrent calls to limit the number of os threads created due to file i/o's (via ledger APIs).
+	if err := e.ThrottleSemaphore.Acquire(ctx); err != nil {
+		return nil, errors.WithMessage(err, "failed to acquire semaphore for endorsement")
+	}
+	defer e.ThrottleSemaphore.Release()
```

Comment:
Both e.Support.ChaincodeEndorsementInfo and e.SimulateProposal call ledger API. I add the throttle in this func right before ChaincodeEndorsementInfo for simplicity. If you prefer, I can refactor the method and move ChaincodeEndorsementInfo and SimulateProposal into a single method.

Reply:
My comment was more about "defer" statement instead of explicitly releasing after line 394.

Reply:
Manish, thank you for the comment. I moved the related code to a separate method because otherwise we had to call Release multiple times due to the error branch.